server/src/main/java/org/elasticsearch/common/blobstore/BlobContainer.java

```diff
@@ -351,6 +357,11 @@ default void compareAndSetRegister(
     /**
      * Gets the value set by {@link #compareAndSetRegister} or {@link #compareAndExchangeRegister} for a given key.
      * If a key has not yet been used, the initial value is an empty {@link BytesReference}.
+     * <p>
+     * This operation has read-after-write consistency with respect to writes performed using {@link #compareAndExchangeRegister} and
+     * {@link #compareAndSetRegister}, but does not guarantee full linearizability. In particular, a {@code getRegister} performed during
+     * one of these write operations may return either the old or the new value, and a caller may therefore observe the old value
+     * <i>after</i> observing the new value, as long as both such read operations take place before the write operation completes.
```

Comment:
Should this comment or part of it be added to `S3BlobContainer` instead? The default implementation usees `compareAndExchangeRegister` which is linearizable?

Reply:
In fact we can remove that default implementation, it's not actually needed any more, see https://github.com/elastic/elasticsearch/pull/131604.